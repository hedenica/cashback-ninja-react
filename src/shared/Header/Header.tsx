import { useCashback, useOrders } from 'resources'
import ninjaImg from 'assets/ninja-cashback.png'

import styles from './header.module.scss'

export function Header () {
  const { data, isFetched, isLoading } = useOrders()
  const { cashbackTotal } = useCashback(isFetched ? data : [])

  const finalCashback = isLoading ? 'XX,XX' : cashbackTotal

  return (
    <div className={styles.container}>
      <header className={styles.header}>
        <div className={styles.title}>
          <img src={ninjaImg} alt='Cashback Ninja Logo' />
          <h1>Cashback Ninja</h1>
        </div>
        <blockquote className={styles.blockquote}>
          "O melhor cashback da internet"
        </blockquote>
        <h3>{`Seu saldo em cashback: ${finalCashback}`}</h3>
      </header>
    </div>
  )
}
