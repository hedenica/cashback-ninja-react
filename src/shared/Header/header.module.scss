.container {
  width: 100%;
  height: 300px;
  background-image: url('../../assets/header-background.svg');
  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
}

.header {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: var(--text);
}

.title {
  display: flex;
  align-items: center;
  letter-spacing: 0.125rem;
  padding: 1.5rem;

  h1 {
    font-size: 2.5rem;
  }

  img {
    width: 60px;
    height: 60px;
    margin-right: 20px;
  }
}

.blockquote {
  letter-spacing: 0.188rem;
  margin: 1.5rem 0;
  font-size: 1.5rem;
}