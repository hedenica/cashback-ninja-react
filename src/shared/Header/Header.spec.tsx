import { QueryClient, QueryClientProvider } from 'react-query'
import { render } from '@testing-library/react'

import { Header } from './Header'

const queryClient = new QueryClient()

describe('Header', () => {
  it('should render properly', () => {
    const { container } = render(
      <QueryClientProvider client={queryClient}>
        <Header />
      </QueryClientProvider>,
    )

    expect(container).toMatchSnapshot()
  })
})
