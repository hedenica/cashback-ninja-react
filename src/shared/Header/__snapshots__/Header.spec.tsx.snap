// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render properly 1`] = `
<div>
  <div
    class="container"
  >
    <header
      class="header"
    >
      <div
        class="title"
      >
        <img
          alt="Cashback Ninja Logo"
          src="ninja-cashback.png"
        />
        <h1>
          Cashback Ninja
        </h1>
      </div>
      <blockquote
        class="blockquote"
      >
        "O melhor cashback da internet"
      </blockquote>
      <h3>
        Seu saldo em cashback: XX,XX
      </h3>
    </header>
  </div>
</div>
`;
